autogpt_platform/backend/backend/integrations/codex/runtime.py

```
@@ -0,0 +1,961 @@
+                futures = self._dynamic_tool_futures.get(call.thread_id)
+                if self.closed or registration is None or futures is None:
+                    return _dynamic_tool_error("codex_tool_handler_unavailable")
+                handler, timeout_seconds = registration
+
+                async def execute_handler() -> CodexDynamicToolResult:
+                    return await handler(call)
+
+                coroutine = execute_handler()
+                try:
+                    future = asyncio.run_coroutine_threadsafe(coroutine, loop)
+                except BaseException:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `BaseException`

(BLE001)